app/redux/ducks/classify.js

```diff
@@ -264,8 +286,9 @@ export function addIntervention(data) {
 }
 
 export function clearIntervention() {
-  return {
-    type: CLEAR_INTERVENTION
+  return (dispatch) => {
+    dispatch({type: STORE_INTERVENTION_UUID});
```

Comment:
I wasn't sure that this was the way to do it, this code path isn't tested in specs by the look. Thoughts appreciated.

Reply:
As someone who isn't familiar with the implementation of interventions in general, reading this is confusing and hard to follow: A clear intervention function is called but then is calling an action to store the id in the state, then in the classification metadata, then the id gets cleared from the state? I would suggest at the very least adding a separate redux action for clearing the UUID if there's going to be an action for storing it. Calling to store the UUID here doesn't seem quite like the right place to me either. The naming of `clearIntervention` would not convey to me that actions are called in the function that do store data.

Reply:
Firing two state changes together should be ok, but I don’t know if Redux guarantees that the new state after saving the UUID will be the initial state for clearing the intervention. 

I’d be inclined to return a single action which remembers the UUID then clears the intervention, so state is only updated once.

Reply:
The specs for these actions are the first tests I wrote for redux in PFE and I'm not sure any more that testing the actions directly is right.

You should be able to test the `clearIntervention()` action creator if you import it and the reducer into the tests, and pass it a mock `dispatch`. Have a look at the specs for the translations action creators, where I used that approach. The mock dispatch function I came up with is this one:
https://github.com/zooniverse/Panoptes-Front-End/blob/9461eb8f83b4b12e0b9bee501cbe8d535783e1a7/app/redux/ducks/translations.spec.js#L26-L31

Reply:
I've pushed a test for the `clearIntervention` action creator.